corehq/apps/users/tasks.py

```diff
@@ -36,6 +39,25 @@ def tag_docs_as_deleted(cls, docs, deletion_id):
     cls.get_db().bulk_save(docs)
 
 
+@task(rate_limit=2, queue='background_queue', ignore_result=True)  # 2 saves/sec for cloudant slowness
+def tag_forms_as_deleted_rebuild_associated_cases(cls, formlist, deletion_id, deleted_cases=set()):
+    from casexml.apps.case.cleanup import rebuild_case
+
+    for doc in formlist:
+        doc['doc_type'] += DELETED_SUFFIX
+        doc['-deletion_id'] = deletion_id
+    cls.get_db().bulk_save(formlist)
+
+    cases_to_rebuild = set()
+    for form in formlist:
+        cases = get_case_ids_from_form(form) - deleted_cases
+        for case in iter_docs(CommCareCase.get_db(), cases):
+            if case['owner_id'] != form['form']['meta']['userID']:
```

Comment:
Thinking about this again I'm not sure this check is necessary since `deleted_cases` should contain ALL cases owned by the user so any other case_id's in the form that aren't in `deleted_cases` must have a different owner.